src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -74,6 +74,9 @@
   // Counter to generate unique request ids per client
   private final AtomicLong counter = new AtomicLong(0);
 
+  // Provides constant values set by Snowflake server
+  private static final ParameterProvider parameterProvider = new ParameterProvider();
```

Comment:
why is this a static variable? What if the value changed between multiple clients creation?

Reply:
Each client should have a single parameterProvider.  ParameterProvider can change its values on each client/configure call without reassigning the Object on the client